### PR TITLE
Install and configure duo

### PR DESCRIPTION
**Pull request by limed**
This will install and configure `duo_unix` using confd

### CODE REVIEW

**Comment by limed** [comment]
Closes issue #54

**Comment by gozer** [comment]
Small nit, make the smart fail domain configurable too.

File: nubis/puppet/duo.pp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+file { '/etc/confd/conf.d/duo.toml':
+    ensure  => 'file',
+    owner   => root,
+    group   => root,
+    mode    => '0644',
+    source  => 'puppet:///nubis/files/duo.toml',
+    require => Package['duo_unix'],
+}
+
+file { '/etc/confd/templates':
```
Comment (gozer) [inline nubis/puppet/duo.pp]: should be:
     file { '/etc/confd/templates/duo.tmpl':

File: nubis/puppet/files/duo.tmpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+pushinfo = yes
+
+; Hack to enable LDAP translation from uid -> mail attribute in ldap
+; since our mail and uid attributes aren't necessarily the same :( :( :(
+ldap_server = {{ if exists "/ldap-server" }}{{ getv "/ldap-server" }}{{ end }}
+ldap_binddn = {{ if exists "/ldap-binddn"}}{{ getv "/ldap-binddn" }}{{ end }}
+ldap_password = {{ if exists "/ldap-bindpassword" }}{{ getv "/ldap-bindpassword" }}{{ end }}
+ldap_basedn = {{ if exists "/ldap-basedn" }}{{ getv "/ldap-basedn" }}{{ end }}
+ldap_resolve_hack = 1
+ldap_smartfail_domain = mozilla.com
```
Comment (gozer) [inline nubis/puppet/files/duo.tmpl]: make configurable too

**Comment by limed** [comment]
And to boot I forgotten to check in my confd fixup script

**Comment by gozer** [comment]
Looking good, only question is what happens if a jumphost is started and none of these consul keys exists?

**Comment by limed** [comment]
Just won't get configured, having the duo not be configured isn't a big deal. It only becomes a big problem when we configure pam

**Comment by tinnightcap** [comment]
r+

**Pull request merged**